### PR TITLE
chaos: assert mock expects in storer tests

### PR DESCRIPTION
**Pull request by danielhochman**
### Description
Tests are not fully verifying functionality.

### Testing Performed
Unit.

### CODE REVIEW

**Comment by danielhochman** [comment]
One of tests also panics locally for me 

```
--- FAIL: TestCreateOrGetExperiment (0.00s)
    storer_test.go:265: 
                Error Trace:    storer_test.go:265
                Error:          Received unexpected error:
                                ExecQuery 'INSERT INTO experiment_config (id, details) VALUES ($1, $2)', arguments do not match: argument 1 expected [string - {"@type":"type.googleapis.com/clutch.chaos.serverexperimentation.v1.HTTPFaultConfig","faultTargeting":{"upstreamEnforcing":{"upstreamCluster":{"name":"upstreamCluster"},"downstreamCluster":{"name":"downstreamCluster"}}},"abortFault":{"percentage":{"percentage":100},"abortStatus":{"httpStatusCode":401}}}] does not match actual [string - {"@type":"type.googleapis.com/clutch.chaos.serverexperimentation.v1.HTTPFaultConfig", "faultTargeting":{"upstreamEnforcing":{"upstreamCluster":{"name":"upstreamCluster"}, "downstreamCluster":{"name":"downstreamCluster"}}}, "abortFault":{"percentage":{"percentage":100}, "abortStatus":{"httpStatusCode":401}}}]
                Test:           TestCreateOrGetExperiment
    --- FAIL: TestCreateOrGetExperiment/1 (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd16de0]

goroutine 43 [running]:
testing.tRunner.func1.2(0xdab420, 0x1364f60)
        /usr/local/go/src/testing/testing.go:1144 +0x49f
testing.tRunner.func1(0xc000582900)
        /usr/local/go/src/testing/testing.go:1147 +0x695
panic(0xdab420, 0x1364f60)
        /usr/local/go/src/runtime/panic.go:971 +0x499
github.com/lyft/clutch/backend/service/chaos/experimentation/experimentstore.TestCreateOrGetExperiment.func1(0xc000582900)
        /home/dhochman/go/src/github.com/lyft/clutch/backend/service/chaos/experimentation/experimentstore/storer_test.go:266 +0x800
testing.tRunner(0xc000582900, 0xc00050e900)
        /usr/local/go/src/testing/testing.go:1194 +0x203
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1239 +0x5d8
```

File: backend/service/chaos/experimentation/experimentstore/storer_test.go [hunk 1]
```diff
@@ -143,6 +143,7 @@ func TestCreateExperiments(t *testing.T) {
 			s := ExperimentSpecification{RunId: "1", ConfigId: "1", StartTime: test.startTime, EndTime: nil, Config: test.config}
 			_, err = es.CreateExperiment(context.Background(), &s)
 			a.Equal(test.err, err)
+			assert.NoError(t, mock.ExpectationsWereMet())
```
Comment (Augustyniak) [inline backend/service/chaos/experimentation/experimentstore/storer_test.go:146]:
```suggestion
			a.NoError(mock.ExpectationsWereMet())
```

File: backend/service/chaos/experimentation/experimentstore/storer_test.go [hunk 2]
```diff
@@ -263,6 +264,7 @@ func TestCreateOrGetExperiment(t *testing.T) {
 			result, err := es.CreateOrGetExperiment(context.Background(), &s)
 			a.NoError(err)
 			a.Equal(tt.expectedOrigin, result.Origin)
+			assert.NoError(t, mock.ExpectationsWereMet())
```
Comment (Augustyniak) [inline backend/service/chaos/experimentation/experimentstore/storer_test.go:267]:
```suggestion
			a.NoError(mock.ExpectationsWereMet())
```

**Review by Augustyniak — APPROVED** [review]
2 suggestions related to consistency in these tests. Thanks for fixing

**Review by Augustyniak — APPROVED** [review]

**Pull request merged**